File: Lesson1/Dice/Dice/Cup.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Dice
{
    internal class Cup
    {
        private List<IDie> dice;

        public Cup()
        {
            this.dice = new List<IDie>();
        }

        public string Roll()
        {
            var total = this.dice.Sum(d => d.Roll());
            return string.Format("You rolled {0} dice which totaled: {1}", dice.Count, total);
        }

        public void Empty()
        {
            this.dice.Clear();
        }

        public void Add(IDie die)
        {
            this.dice.Add(die);
        }
    }
}

File: Lesson1/src/main/java/com/mikeabney/katas/dice/domain/Cup.java
package com.mikeabney.katas.dice.domain;

import java.util.HashSet;
import java.util.Set;

public class Cup {
    Set<Die> dice;

    public Cup() {
        dice = new HashSet<>();
    }

    public int roll() {
        int total = 0;
        for (Die die : dice) {
            total += die.roll();
        }
        return total;
    }

    public void addDie(Die die) {
        dice.add(die);
    }

    public void empty() {
        dice.clear();
    }

    public int size() {
        return dice.size();
    }
}
